liquidctl/driver/nzxt_smart_device.py

```diff
@@ -182,66 +276,244 @@ def get_status(self, **kwargs):
+            LOGGER.debug('received %s', ' '.join(format(i, '02x') for i in msg))
+            if not msg_1101_reply and msg[0] == 0x11 and msg[1] == 0x01:
+                fw = '{}.{}.{}'.format(msg[0x11], msg[0x12], msg[0x13])
+                status.append(('Firmware version', fw, ''))
+                num_valid_replies_recvd += 1
+                msg_1101_reply = True
+                continue
+            if not msg_2103_reply and msg[0] == 0x21 and msg[1] == 0x03:
+                num_light_channels = msg[14]  # the 15th byte (index 14) is # of light channels
+                accessories_per_channel = 6   # each lighting channel supports up to 6 accessories
+                light_accessory_index = 15    # offset in msg of info about first light accessory
+                for light_channel in range(num_light_channels):
+                    for accessory_num in range(accessories_per_channel):
+                        accessory_id = msg[light_accessory_index]
+                        light_accessory_index += 1
+                        if accessory_id != 0:
+                            status.append(('LED {} accessory {}'.format(light_channel + 1, accessory_num + 1),
+                                           self._ACCESSORY_NAMES.get(accessory_id, 'Unknown'), ''))
+                num_valid_replies_recvd += 1
+                msg_2103_reply = True
+                continue
+            if not msg_6702_reply and msg[0] == 0x67 and msg[1] == 0x02:
+                rpm_offset = 24
+                duty_offset = 40
+                noise_offset = 56
+                for i, _ in enumerate(self._speed_channels):
+                    if ((msg[rpm_offset] != 0x0) and (msg[rpm_offset + 1] != 0x0)):
+                        status.append(('Fan {} speed'.format(i + 1), msg[rpm_offset + 1] << 8 | msg[rpm_offset], 'rpm'))
+                        status.append(('Fan {} duty'.format(i + 1), msg[duty_offset + i], '%'))
+                    rpm_offset += 2
+                status.append(('Noise level', msg[noise_offset], 'dB'))
+                num_valid_replies_recvd += 1
+                msg_6702_reply = True
+                continue
+            if not msg_6704_reply and msg[0] == 0x67 and msg[1] == 0x04:
+            # PLACE HOLDER FOR FUTURE WORK
+            # Interpretation of this reply is pending
+                #status.append((' Unknown:', 'status', 'pending'))
+                num_valid_replies_recvd += 1
+                msg_6704_reply = True
+                continue
+        self.device.release()
+        return sorted(status)
+
+    def _write_colors(self, cid, mode, colors, sval):
+        mval, mod3, mod4, mincolors, maxcolors = self._COLOR_MODES[mode]
+        color_count = len(colors)
+        if maxcolors == 40:
+            channel_mod = [0x1E, 0x20][cid] # the purpose of this is unknown, but is based on cmd issued by CAM software
+            led_padding = [0xFF, 0xFF, 0xFF]*(maxcolors - color_count)  # set all remaining LEDs to White
```

Comment:
This custom padding is probably not required, it should just as well work with the default 0x00 padding inserted by _write.

Reply:
This is just a matter of preference. These are not space-padding bytes. Instead, they actually set the remaining LEDs to white. If the user specifies 5 out of 40 LEDs, for example, then the remaining ones will be set to white. We could turn off the remaining one by setting `led_padding` to 0x00 -- hence this is a matter of preference. CAM Software sets them to White.

Reply:
It's true that it's a matter of preference.

I'm not ready to die on this hill, but I will argue that off – which is what RGB(0, 0, 0) results in – might be a better default for these modes: the user only has to specify colors for the LEDs it wants lit, and the remaining are left off.  This is inline with the docs: number of colors between 1 and 40.

---

If I managed to convince you with this argument, great : )

Otherwise, feel free to leave it as is.  I might change it in the future for consistency with the other driver, but I could also end up agreeing with you and changing the other drivers.

Reply:
I really don't mind either way. I was mimicking CAM. I've changed it to 0x00, but kept the code as-is in order to accommodate future revision. For example, a forwarded parameter could be used to specify the color for the remaining LEDs.
```Python
led_padding = [0x00, 0x00, 0x00]*(maxcolors - color_count)  # set all remaining LEDs to Black (we might change this in the future)
```

Reply:
Fair enough.